### PR TITLE
Wait for sky server to start before starting device

### LINKED ISSUE #141: Race condition with starting HTTP server
"I think we figured it out.  It looks like it was a race condition between stating the local HTTP server and telling the device to load main.dart."

### PR DESCRIPTION
**Pull request by tvolkert**
Instead of just waiting for the sky server process to start
before we start the activity on the device, this causes us to
wait for the sky server to actually start listening on its port

Fixes #141

### CODE REVIEW

File: lib/src/device.dart [hunk 1]
```diff
@@ -512,6 +513,9 @@ class AndroidDevice extends Device {
   static const String className = 'AndroidDevice';
   static final String defaultDeviceID = 'default_android_device';
 
+  static const String _SKY_SERVER_START_MSG = 'Serving';
+  static const Duration _SKY_SERVER_TIMEOUT = const Duration(seconds: 3);
```
Comment (abarth) [inline lib/src/device.dart]: The current Dart style is to use camelCase for constants.  The old style was to use SHOUTING_CASE, which is why you've probably seen examples of that in the Dart SDK code.

File: lib/src/device.dart [hunk 2]
```diff
@@ -773,8 +777,14 @@ class AndroidDevice extends Device {
           [adbPath, 'forward', observatoryPortString, observatoryPortString]);
 
       // Actually start the server.
-      await Process.start(sdkBinaryName('pub'), ['run', 'sky_tools:sky_server', _serverPort],
-          workingDirectory: serverRoot, mode: ProcessStartMode.DETACHED);
+      Process ps = await Process.start(
```
Comment (abarth) [inline lib/src/device.dart]: ps <--- please use complete words for variable names...  Maybe `server` ?

**Comment by abarth** [comment]
LGTM

File: lib/src/device.dart (hunk 1 above)
Comment (Hixie) [inline lib/src/device.dart]: in general we use `kFooBar` style, as in `kSkyServerStartMessage`.

Also, don't use abbreviations ("message" not "msg").
Also, use "flutter" not "sky" where possible.

Comment (tvolkert) [inline lib/src/device.dart]: Done

File: lib/src/device.dart (hunk 2 above)
Comment (tvolkert) [inline lib/src/device.dart]: Done

**Pull request merged**